File: apps/mobile-flutter/lib/core/utils/ids.dart
/// Generates locally unique ids for new records.
///
/// There is no backend this term, so a timestamp plus a per-launch counter is
/// enough to keep ids unique inside one device's `SharedPreferences`.
class IdGenerator {
  IdGenerator({DateTime Function()? now}) : _now = now ?? DateTime.now;

  final DateTime Function() _now;
  int _counter = 0;

  String next(String prefix) {
    _counter += 1;
    return '$prefix-${_now().microsecondsSinceEpoch}-$_counter';
  }
}

File: apps/react-mobile/src/core/utils/ids.ts
/**
 * Generates locally unique ids for new records.
 *
 * There is no backend this term, so a timestamp plus a per-instance counter
 * is enough to keep ids unique inside one device's AsyncStorage.
 *
 * Port of lib/core/utils/ids.dart.
 */
export class IdGenerator {
  private counter = 0;
  private readonly now: () => number;

  constructor(now: () => number = () => Date.now()) {
    this.now = now;
  }

  next(prefix: string): string {
    this.counter += 1;
    return `${prefix}-${this.now()}-${this.counter}`;
  }
}
